src/OctostacheCmd/EnvironmentVariableRetriever.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OctostacheCmd
+{
+    public class EnvironmentVariableRetriever
+    {
+        private static string[] _predefinedVariablePrefixes =
+            {"runner.", "azure_http_user_agent", "common.", "system."};
```

Comment:
What are these for exactly?

Reply:
They can probably be removed to be honest. But I was looking at [microsoft's variable-substitution action](https://github.com/microsoft/variable-substitution/blob/master/src/operations/envVariableUtility.ts#L2) and they were using these to ostensibly prevent unwanted substitutions or maybe to lower the amount of variables they loop through. Either way that's probably not a problem for us so I can remove this stuff.

Reply:
I don't think you have to remove it I was just wondering what is was for.  Is it to help weed out env variables that live on the machine?

Reply:
Too late, I already removed it! But yeah, that's what I was thinking it was for.